packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -59,13 +66,58 @@ export async function tryHotswapDeployment(
 
 async function findAllHotswappableChanges(
   stackChanges: cfn_diff.TemplateDiff, evaluateCfnTemplate: EvaluateCloudFormationTemplate,
+  sdk: ISDK, rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedStackNames: {[key: string]: any}, hotswappableResources?: HotswapOperation[],
 ): Promise<HotswapOperation[] | undefined> {
   const resourceDifferences = getStackResourceDifferences(stackChanges);
 
   let foundNonHotswappableChange = false;
   const promises: Array<Array<Promise<ChangeHotswapResult>>> = [];
+
+  if (!hotswappableResources) {
+    hotswappableResources = new Array<HotswapOperation>();
+  }
+
+  // process any resources in nested stacks
+  // if any of them are not hotswappable, fail now
+  const resourceDifferenceEntries = Object.entries(resourceDifferences);
+  for (const [logicalId, change] of resourceDifferenceEntries.filter(
+    ([_, resourceDifference]) => (resourceDifference.newValue?.Type === 'AWS::CloudFormation::Stack' && resourceDifference.oldValue?.Type === 'AWS::CloudFormation::Stack'))
+  ) {
+    const parameters = change.newValue?.Properties?.Parameters;
+    if (parameters) {
+      for (const paramName in parameters) {
+        if (typeof parameters[paramName] !== 'string') {
+          parameters[paramName] = await evaluateCfnTemplate.evaluateCfnExpression(parameters[paramName]);
+
+        }
+      }
+    }
```

Comment:
I would rewrite this to a single `reduce` statement:

```suggestion
    const parameters = Object.entries(change.newValue?.Properties?.Parameters ?? {}).reduce(async (acc, [paramName, paramValue]) => {
      acc[paramName] = await evaluateCfnTemplate.evaluateCfnExpression(paramValue);
      return acc;
    });
```

Reply:
Actually, this can be simplified to just

```suggestion
    const parameters = await evaluateCfnTemplate.evaluateCfnExpression(change.newValue?.Properties?.Parameters);
```

, no?